### PR TITLE
Add `yii\rbac\ManagerInterface::hasAssignments()`

### PR DESCRIPTION
**Pull request by Kolyunya**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | yes
| Tests pass?   | yes
| Fixed issues  | #8748

### CODE REVIEW

File: framework/rbac/ManagerInterface.php [hunk 1]
```diff
@@ -230,6 +230,13 @@ public function getAssignments($userId);
     public function getUserIdsByRole($roleName);
 
     /**
+     * Returns a boolean indicating whether the role has assignments.
+     * @param string $roleName the role name
+     * @return bool whether the role has assignments.
```
Comment (samdark) [inline framework/rbac/ManagerInterface.php]: Need `@since 2.0.12`

**Review by samdark — CHANGES_REQUESTED** [review]

**Comment by samdark** [comment]
- Need a line for changelog.
- The change should be mentioned in UPGRADE since it may be breaking for custom RBAC manager implementations (necessary to implement additional method in this case).

**Comment by Kolyunya** [comment]
@samdark I used to think that we can not break the BC in minor releases/ If we indeed can then why is #11927 assigned to `2.1.x`? I used to think that it's because of the BC break.

**Comment by samdark** [comment]
Right. Assigned to 2.1.0.

**Comment by Kolyunya** [comment]
@samdark I've just checked the `UPGRADE.md` and noticed that it contains many BC breaks like:

> * RBAC: `getChildRoles()` method was added to `\yii\rbac\ManagerInterface`. If you've implemented your own RBAC manager
  you need to implement new method.

That's very confusing... If we already have breaking changes then why can't we introduce new ones?

**Comment by samdark** [comment]
We're getting more strict on versions.

**Pull request closed without merging**